sqlmesh/core/model/meta.py

```diff
@@ -37,7 +39,9 @@ class ModelMeta(Node):
     column_descriptions_: t.Optional[t.Dict[str, str]]
     audits: t.List[AuditReference] = []
     tags: t.List[str] = []
-    grain: t.List[str] = []
+    grain: t.Optional[Reference] = None
+    grains: t.List[Reference] = []
```

Comment:
I don't think we should be doing this. I think we should only keep `grains` and then `pop` the `grain` key in the `root_validator` and set the `grains` one instead. I'm not sure I like having 2 versions in the model definition itself.